### DISCUSSION TITLE
Unable to access an image in the Public folder of my webapp

### DISCUSSION THREAD

**Discussion by zhao-binxiang** [vercel/next.js#57096] [Help] [answered]
### Summary

I saved an image file to the public folder
`      const resizedImage = await sharp(imageBuffer)
        .composite(layers)
        .toFile("./public/final.png");`

It builds successfully but when I run the code, it gives me the following error
`- error [Error: public/final.png: unable to open for write
unix error: No such file or directory`

What shall I do differently?

**Comment by zhao-binxiang** [comment]
I should say that the webapp runs fine locally. And the directory/files do exist...

**Comment by icyJoseph** [comment, marked as answer]
I don't think you can write to the public folder on Vercel. 

- https://github.com/orgs/vercel/discussions/241#discussioncomment-363002

You also might want to use `process.cwd()` and path to actually locate `./public/final.png`. During runtime serverless functions run from a different directory, so `process.cwd()` is necessary to find the right path.

- https://nextjs.org/docs/pages/api-reference/functions/get-static-props#reading-files-use-processcwd

**Comment by zhao-binxiang** [comment]
Yeah. My bad. I decided to use Blob to handle the images. Thanks
